### ISSUE TITLE
low priority user `JsonFormat[T]` will not stop derivation of a `familyFormat[T]`

### ISSUE THREAD

**Issue by fommil** [fommil/spray-json-shapeless#8]
https://github.com/fommil/spray-json-shapeless/commits/xor

I thought I'd fixed this with the `Xor` (ambiguous implicits trick), but apparently not. It all seems very fragile.

@milessabin new ticket for this.

**Comment by milessabin** [comment]
OK, I have this as a fairly unsatisfactory workaround: https://github.com/milessabin/spray-json-shapeless/commit/5a8bc59dd005688383c3a110eb838716e0a57fa5. I think for the time being that's about as good as it's going to get.

One major obstacle to improving the situation is that all the implicits (both yours and the spray-json ones) are brought into scope via imports. This has two consequences, neither of which are widely understood,

1. They will always be used in preference to implicits defined in the implicit scope.
2. Implicit prioritization only applies to implicits if they are resolved via the implicit scope rather than by imports or any other form of local accessibility.

Together they imply that basically all of your low priority trait stacking is useless.

I think your best bet would be to start by dramatically simplifying things in the light of the above, and then maybe we can revisit this when @alexarchambault's `Lazy` improvements have landed.

**Comment by fommil** [comment]
heh, interesting. This is starting to sound like exactly the same problem as #1 and can also be worked around by redefining the implicit val in the `ExampleFormats` (like we do for `Symbol`).

Thanks for looking into it! I think I will remove the `Xor` for 1.1.

**Comment by fommil** [comment]
@milessabin I'm going to make a release soon with the removal of `Xor` but I can't lose the low priority hierarchy of `FamilyFormat`s because `optionFormat` needs to be higher priority when compiling. Any ideas on that? It would be good to reduce this to one trait.

**Comment by milessabin** [comment]
Did you try flattening that out completely? You're directly importing, so I don't believe that there is any implicit prioritization there at all.

**Comment by fommil** [comment]
I remove `LowPriorityFamilyFormats` and just had `FamilyFormats`. Did you have something else in mind?

**Comment by milessabin** [comment]
Sorry, ignore me. You're depending on the standard overload resolution rules here. Makes sense.

**Comment by fommil** [comment]
I'll probably make a release today/tomorrow with this change (plus license), any news on the derived orphans stuff or are we waiting for the better Lazy?

**Comment by milessabin** [comment]
The way that spray-json is structured means that none of the mechanisms for handling implicit priority in shapeless are going to help. So, yes, I think we're waiting on improvements to `Lazy`.
